### ISSUE TITLE
Missing database host column

### ISSUE THREAD

**Issue by bjorn2404** [No3x/wp-mail-logging#52]
Hello,

I just noticed this plugin stopped working for me in November and it seems to be due to not having a 'host' column in the ..._wpml_mails database table. Error:

WordPress database error Unknown column 'host' in 'field list' for query INSERT INTO...

The last working log in my database appears to be from version 1.6.2 of the plugin.

**Comment by No3x** [comment]
Hi @bjorn2404,
unfortunately this is a know issue to me. This happens in rare cases but It did not occur on any of my websites I use the plugin on my own. 
I fixed this in 1.8.0 (I hope I actually did) which will be released soon (as soon as I receive a missing translation see #51 ).

Fortunately there are multiple ways to fix this:
the easiest one is to enable cleanup in the settings and then re-enable the plugin. This will setup the options and the plugin again. But you will loose your logs of course.
If this is no option for you please let me know to find a better strategy.

**Comment by bjorn2404** [comment]
I don't have a problem manually creating the column in my database but can't you just add a check for the column? Or maybe that's what you'll do with 1.8.0. I updated the plugin with the updater.

```php
/**
 * Check to see if the database column exists.
 */
public function check_db_column() {
	global $wpdb;
	$tableName = $this->getTablename( 'mails' );

	$results = $wpdb->get_results( $wpdb->prepare( "SHOW COLUMNS FROM `$tableName` LIKE %s", 'host' ) );
	$column_exists = ( count( $results ) > 0 ) ? true : false;

	if ( false === $column_exists && is_array( $results ) ) {
		$wpdb->query("ALTER TABLE `$tableName` ADD COLUMN `host` VARCHAR(800) NOT NULL DEFAULT '0' AFTER `timestamp`;");
	}
}
```

**Comment by No3x** [comment]
I did not think a lot of people were affected by this because there are few cases about it compared to the time 1.7.0 is out. 
So I just improved the upgrade process that the upgrade should not fail like this again. I did not added something to fix the database for 1.7.0 affected users in version 1.8.0 - but I definitely should. I will use your fix that will run on 1.8.0 database upgrade.

Thanks for the hint! I really appreciate it.
